lib/vanagon/engine/base.rb

```diff
@@ -1,11 +1,55 @@
 require 'vanagon/utilities'
 require 'vanagon/errors'
+require 'vanagon/engine'
 
 class Vanagon
   class Engine
+    # Base is the Engine subclass that all other Engines subclasses should inherit from.
+    # It defines the general interface of an Engine, and gives them the ability to
+    # register themselves with the Vanagon::Engine class. Without inheriting from Base,
+    # a new Engine has to completely reimplement the loosely binding Constract of
```

Comment:
![](http://2new1.fjcdn.com/reaction_gifs/English+can+either+be+difficult+or+too+hard+for+someone+_84f9f93d730fbb72a4ce6a42889670b3.gif)